### PR TITLE
Test

### PR DESCRIPTION
**Pull request by Nandakishore19**
Question, Answer model

### CODE REVIEW

File: quorabase/models.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+from django.db import models
+from django.contrib.auth.models import User
+from django.utils import timezone
+from vote.models import VoteModel
```
Comment (rohith2931) [inline quorabase/models.py]: Where is the `vote` app installed?
Can we achieve voting functionality using many to many relation and a intermediate table?

Comment (Nandakishore19) [reply quorabase/models.py]: Yes, we can achieve voting functionality by using an intermediate table. I haven't reached the voting part of my app yet, so I haven't updated it.I found this VoteModel, which makes it easy to implement a voting system.

File: quorabase/models.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+from django.urls import reverse
+
+
+# Create your models here.
+class Question(models.Model):  # Table name quorabase_question
```
Comment (rohith2931) [inline quorabase/models.py:7]: add `description` field to the Question Model

File: QuoraClone/settings.py [hunk 3]
```diff
@@ -31,52 +32,57 @@
...
+    "users.apps.UsersConfig",
+    "django.contrib.admin",
+    "django.contrib.auth",
+    "django.contrib.contenttypes",
+    "django.contrib.sessions",
+    "django.contrib.messages",
+    "django.contrib.staticfiles",
+    "crispy_forms",
+    "crispy_bootstrap4",
+    "vote",
```
Comment (rohith2931) [inline QuoraClone/settings.py]: remove un-used apps like `vote` here.

File: quorabase/views.py [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+    def test_func(self):
+        answer = self.get_object()
+
+        return self.request.user == answer.author
+
+
+@login_required
+def upvote_question(request, question_id):
+    question = get_object_or_404(Question, pk=question_id)
```
Comment (rohith2931) [inline quorabase/views.py]: handle exception

File: quorabase/views.py [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+    if user in question.vote.all():
+        question.vote.remove(user)
+    else:
+        question.vote.add(user)
+    return redirect("quora-home")
+
+
+@login_required
+def upvote_answer(request, question_id, answer_id):
+    answer = get_object_or_404(Answer, pk=answer_id)
```
Comment (rohith2931) [inline quorabase/views.py]: handle exception here as well

File: quorabase/views.py [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+    user = request.user
+    if user in question.vote.all():
+        question.vote.remove(user)
+    else:
+        question.vote.add(user)
+    return redirect("quora-home")
+
+
+@login_required
+def upvote_answer(request, question_id, answer_id):
```
Comment (rohith2931) [inline quorabase/views.py]: remove unused `question_id` parameter

File: quorabase/urls.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        name="answer-update",
+    ),
+    path(
+        "question/<int:question_pk>/answer/<int:answer_pk>/delete/",
+        AnswerDeleteView.as_view(),
+        name="answer-delete",
+    ),
+    path("question/upvote/<int:question_id>", upvote_question, name="upvote-question"),
+    path(
+        "question/<int:question_id>/answer/<int:answer_id>/upvote/",
```
Comment (rohith2931) [inline quorabase/urls.py]: change the url path to `answer/upvote/<int:answer_id>/`

**Pull request merged**